lib/query.js

```diff
@@ -72,4 +116,112 @@ export const getDenominator = (node) => {
     } else {
         return null
     }
-}
\ No newline at end of file
+}
+
+// TODO: handle multivariable polynomials
+// Get degree of a polynomial term
+// e.g. 6x^2 -> 2
+export const getPolyDegree = (node) => {
+    if (isNumber(node)) {
+        return build.number(0)
+    } else if (isIdentifier(node) || isPolynomial(node)){
+        return build.number(1)
+    } else if (isPow(node)) {
+        return node.args[1]
+    } else if (isMul(node)){
+        return getPolyDegree(node.args[1])
+    } else if (isNeg(node)) {
+        const variable = node.args[0]
+        return getPolyDegree(variable.args[1])
+    } else {
+        return null
+    }
+}
+
+export const getCoefficient = (node) => {
+    if (isNumber(node)) {
+        return node
+    } else if (isIdentifier(node) || isPow(node)) {
+        return build.numberNode('1')
+    } else if (isNeg(node)) {
+        const result = build.applyNode('neg', [getCoefficient(node.args[0])])
+        result.wasMinus = node.wasMinus
+        return result
+    } else if (isMul(node)) {
+        const numbers = node.args.filter(arg => isNumber(arg) || isConstantFraction(arg))
+        if (numbers.length > 1) {
+            return build.mul(...numbers)
+        } else if (numbers.length > 0) {
+            return numbers[0]
```

Comment:
The condition looks a little weird... I know that `> 1` has already been checked, but I think that doing
```
} else if (numbers.length === 1) {
   return numbers[0]
} ...
```
would be clearer.